Since v0.22.0, Visual Studio 2022 can't generate cmake cahce.

I'm using CMake Project in Visual Studio 2022.
If I upgrade 'porject_options' to 0.22.0 from 0.21.1, it reports below error and cmake cache generation has failed.

```
1> [CMake] -- Using Windows MSVC toolchain
1> [CMake] -- Setting CXX/C compiler to C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
1> [CMake] -- Running `C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Auxiliary/Build/vcvarsall.bat x64` to set up the MSVC environment
1> [CMake] ERROR: The system was unable to find the specified registry key or value.
1> [CMake] -- Using Windows MSVC toolchain
1> [CMake] -- Setting CXX/C compiler to C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe
1> [CMake] -- Running `C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Auxiliary/Build/vcvarsall.bat x64` to set up the MSVC environment
1> [CMake] ERROR: The system was unable to find the specified registry key or value.
```

weired. I got some warnings on sanitizers when I build now.
Since 'Warning as error' has turned on so compile has failed.
It didn't happens when I used 0.21.1

```
>------ Rebuild All started: Project: Game, Configuration: windows-msvc-debug-hard-mode ------
  [1/1] Cleaning all built files...
  Cleaning... 0 files.
  [1/14] Building CXX object fuzz_test\CMakeFiles\fuzz_tester.dir\fuzz_tester.cpp.obj
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\cl : Command line warning D9002: ignoring unknown option '/fsanitize=fuzzer'
  [2/14] Linking CXX executable fuzz_test\fuzz_tester.exe
  FAILED: fuzz_test/fuzz_tester.exe 
  cmd.exe /C "cd . && "C:\Program Files\Microsoft Visual Studio\2022\Professional\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E vs_link_exe --intdir=fuzz_test\CMakeFiles\fuzz_tester.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe /nologo fuzz_test\CMakeFiles\fuzz_tester.dir\fuzz_tester.cpp.obj  /out:fuzz_test\fuzz_tester.exe /implib:fuzz_test\fuzz_tester.lib /pdb:fuzz_test\fuzz_tester.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console  /fsanitize=address /fsanitize=fuzzer -LIBPATH:C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\ATLMFC\lib\x64   -LIBPATH:C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\lib\x64   -LIBPATH:C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\lib\x86\store\REFERE~1   -LIBPATH:C:\PROGRA~2\WI3CF2~1\10\lib\100190~1.0\ucrt\x64   -LIBPATH:C:\PROGRA~2\WI3CF2~1\10\lib\100190~1.0\um\x64   -LIBPATH:"C:\Program Files (x86)\Windows Kits\10\References\x64" vcpkg_installed\x64-windows\debug\lib\fmtd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\fuzz_test && C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -noprofile -executionpolicy Bypass -file C:/dev/vcpkg/scripts/buildsystems/msbuild/applocal.ps1 -targetBinary E:/work/Knuckles/Game/out/build/windows-msvc-debug-hard-mode/fuzz_test/fuzz_tester.exe -installedDir E:/work/Knuckles/Game/out/build/windows-msvc-debug-hard-mode/vcpkg_installed/x64-windows/debug/bin -OutVariable out""
  LINK Pass 1: command "C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe /nologo fuzz_test\CMakeFiles\fuzz_tester.dir\fuzz_tester.cpp.obj /out:fuzz_test\fuzz_tester.exe /implib:fuzz_test\fuzz_tester.lib /pdb:fuzz_test\fuzz_tester.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console /fsanitize=address /fsanitize=fuzzer -LIBPATH:C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\ATLMFC\lib\x64 -LIBPATH:C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\lib\x64 -LIBPATH:C:\PROGRA~2\MICROS~4\2019\PROFES~1\VC\Tools\MSVC\1429~1.301\lib\x86\store\REFERE~1 -LIBPATH:C:\PROGRA~2\WI3CF2~1\10\lib\100190~1.0\ucrt\x64 -LIBPATH:C:\PROGRA~2\WI3CF2~1\10\lib\100190~1.0\um\x64 -LIBPATH:C:\Program Files (x86)\Windows Kits\10\References\x64 vcpkg_installed\x64-windows\debug\lib\fmtd.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:fuzz_test\CMakeFiles\fuzz_tester.dir/intermediate.manifest fuzz_test\CMakeFiles\fuzz_tester.dir/manifest.res" failed (exit code 1120) with the following output:
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\LINK : warning LNK4044: unrecognized option '/fsanitize=address'; ignored
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\LINK : warning LNK4044: unrecognized option '/fsanitize=fuzzer'; ignored
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\LINK : warning LNK4300: ignoring '/INCREMENTAL' because input module contains ASAN metadata
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\LINK : warning LNK4044: unrecognized option '/fsanitize=address'; ignored
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\LINK : warning LNK4044: unrecognized option '/fsanitize=fuzzer'; ignored
     Creating library fuzz_test\fuzz_tester.lib and object fuzz_test\fuzz_tester.exp
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\LIBCMTD.lib(exe_main.obj) : error LNK2019: unresolved external symbol main referenced in function "int __cdecl invoke_main(void)" (?invoke_main@@YAHXZ)
E:\work\Knuckles\Game\out\build\windows-msvc-debug-hard-mode\fuzz_test\fuzz_tester.exe : fatal error LNK1120: 1 unresolved externals
  [3/14] Building CXX object knuckles\test\constexpr\CMakeFiles\my_lib_relaxed_constexpr_tests.dir\constexpr_tests.cpp.obj
  [4/14] Building CXX object knuckles\test\CMakeFiles\my_lib_tests.dir\tests.cpp.obj
  [5/14] Building CXX object console_app\test\CMakeFiles\my_exe_helpers_tests.dir\tests.cpp.obj
  [6/14] Building CXX object knuckles\test\constexpr\CMakeFiles\my_lib_constexpr_tests.dir\constexpr_tests.cpp.obj
  [7/14] Building CXX object console_app\CMakeFiles\knuckles_exe.dir\src\main.cpp.obj
  [8/14] Building CXX object knuckles\CMakeFiles\knuckles_lib.dir\src\lib.cpp.obj
  ninja: build stopped: subcommand failed.

Rebuild All failed.
```

--- a/src/VCEnvironment.cmake
+++ b/src/VCEnvironment.cmake
@@ -115,12 +115,13 @@ macro(run_vcvarsall)
       message(STATUS "Running `${VCVARSALL_FILE} ${VCVARSALL_ARCH}` to set up the MSVC environment")
       execute_process(
         COMMAND
+          "set" "VSCMD_DEBUG=0" "&&" # make vcvarsall quiet
           "cmd" "/c" "${VCVARSALL_FILE}" "${VCVARSALL_ARCH}" "1>NUL" #
-          "&&" "call" "echo" "VCVARSALL_ENV_START" #
-          "&" "set" #
+          "&&" "call" "echo" "VCVARSALL_ENV_START" # a starting point
+          "&" "set" # print the environment variables
         OUTPUT_VARIABLE VCVARSALL_OUTPUT
         ERROR_VARIABLE VCVARSALL_ERROR
-        OUTPUT_STRIP_TRAILING_WHITESPACE)
+        OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_STRIP_TRAILING_WHITESPACE)
 
       if("${VCVARSALL_ERROR}" STREQUAL "")
         # parse the output and get the environment variables string

--- a/src/VCEnvironment.cmake
+++ b/src/VCEnvironment.cmake
@@ -115,7 +115,7 @@ macro(run_vcvarsall)
       message(STATUS "Running `${VCVARSALL_FILE} ${VCVARSALL_ARCH}` to set up the MSVC environment")
       execute_process(
         COMMAND
-          "cmd" "/c" ${VCVARSALL_FILE} ${VCVARSALL_ARCH} #
+          "cmd" "/c" "${VCVARSALL_FILE}" "${VCVARSALL_ARCH}" "1>NUL" #
           "&&" "call" "echo" "VCVARSALL_ENV_START" #
           "&" "set" #
         OUTPUT_VARIABLE VCVARSALL_OUTPUT

--- a/src/VCEnvironment.cmake
+++ b/src/VCEnvironment.cmake
@@ -130,6 +130,11 @@ macro(run_vcvarsall)
       # recover VSCMD_DEBUG variable
       set($ENV{VSCMD_DEBUG} "${VSCMD_DEBUG}")
 
+      if("${VCVARSALL_ERROR}" STREQUAL ""
+         AND NOT
+             "${VCVARSALL_OUTPUT}"
+             STREQUAL
+             "")
         # parse the output and get the environment variables string
         find_substring_by_prefix(VCVARSALL_ENV "VCVARSALL_ENV_START" "${VCVARSALL_OUTPUT}")
 

--- a/src/VCEnvironment.cmake
+++ b/src/VCEnvironment.cmake
@@ -119,13 +119,20 @@ macro(run_vcvarsall)
           "&&" "call" "echo" "VCVARSALL_ENV_START" #
           "&" "set" #
         OUTPUT_VARIABLE VCVARSALL_OUTPUT
+        ERROR_VARIABLE VCVARSALL_ERROR
         OUTPUT_STRIP_TRAILING_WHITESPACE)
 
-      # parse the output and get the environment variables string
-      find_substring_by_prefix(VCVARSALL_ENV "VCVARSALL_ENV_START" "${VCVARSALL_OUTPUT}")
+      if("${VCVARSALL_ERROR}" STREQUAL "")
+        # parse the output and get the environment variables string
+        find_substring_by_prefix(VCVARSALL_ENV "VCVARSALL_ENV_START" "${VCVARSALL_OUTPUT}")
 
-      # set the environment variables
-      set_env_from_string("${VCVARSALL_ENV}")
+        # set the environment variables
+        set_env_from_string("${VCVARSALL_ENV}")
+      else()
+        message(WARNING "Failed to parse the vcvarsall output. ${VCVARSALL_ERROR}.\nIgnoring this error")
+
+      endif()
+
     else()
       message(
         WARNING

--- a/src/VCEnvironment.cmake
+++ b/src/VCEnvironment.cmake
@@ -113,9 +113,13 @@ macro(run_vcvarsall)
     if(EXISTS ${VCVARSALL_FILE})
       # run vcvarsall and print the environment variables
       message(STATUS "Running `${VCVARSALL_FILE} ${VCVARSALL_ARCH}` to set up the MSVC environment")
+
+      # make vcvarsall quiet
+      set(VSCMD_DEBUG "$ENV{VSCMD_DEBUG}")
+      set($ENV{VSCMD_DEBUG} 0)
+
       execute_process(
         COMMAND
-          "set" "VSCMD_DEBUG=0" "&&" # make vcvarsall quiet
           "cmd" "/c" "${VCVARSALL_FILE}" "${VCVARSALL_ARCH}" "1>NUL" #
           "&&" "call" "echo" "VCVARSALL_ENV_START" # a starting point
           "&" "set" # print the environment variables
@@ -123,7 +127,9 @@ macro(run_vcvarsall)
         ERROR_VARIABLE VCVARSALL_ERROR
         OUTPUT_STRIP_TRAILING_WHITESPACE ERROR_STRIP_TRAILING_WHITESPACE)
 
-      if("${VCVARSALL_ERROR}" STREQUAL "")
+      # recover VSCMD_DEBUG variable
+      set($ENV{VSCMD_DEBUG} "${VSCMD_DEBUG}")
+
         # parse the output and get the environment variables string
         find_substring_by_prefix(VCVARSALL_ENV "VCVARSALL_ENV_START" "${VCVARSALL_OUTPUT}")
 

--- a/src/VCEnvironment.cmake
+++ b/src/VCEnvironment.cmake
@@ -96,7 +96,7 @@ macro(run_vcvarsall)
 
   # if msvc_found is set by msvc_toolchain
   # or if MSVC but VSCMD_VER is not set, which means vcvarsall has not run
-  if(MSVC_FOUND OR (MSVC AND "$ENV{VSCMD_VER}" STREQUAL ""))
+  if((MSVC_FOUND OR MSVC) AND "$ENV{VSCMD_VER}" STREQUAL "")
 
     # find vcvarsall.bat
     get_filename_component(MSVC_DIR ${CMAKE_CXX_COMPILER} DIRECTORY)

